### DISCUSSION TITLE
It would be nice to have some helpers to use IXLRow / IXLColumn objects with .Cell() methods

### DISCUSSION THREAD

**Discussion by airbreather** [ClosedXML/ClosedXML#1966] [Ideas]
(edit to add:) I'm on 0.95.4 and I haven't checked to see if any newer version has actually added these... heh.  I still haven't checked, I'm just editing this on.

I wasn't going to bother with this, but [the wiki page](https://github.com/ClosedXML/ClosedXML/wiki/Simplifying-your-life) does suggest opening a discussion like this if I come up with an idea, so here goes...

```csharp
IXLWorksheet someWorksheet = /* something I've gotten from elsewhere */;
IXLRow someRow = /* something I've gotten from elsewhere, known to be within someWorksheet */;
IXLColumn someColumn = /* something I've gotten from elsewhere, known to be within someWorksheet */;

// it would be nice to be able to do this:
{
    IXLCell cellInWorksheet = someWorksheet.Cell(someRow, someColumn);
    // ^ compiler error (or a different overload) if I switch the order of these parameters

    IXLCell cellInRow = someRow.Cell(someColumn);
    IXLCell cellInColumn = someColumn.Cell(someRow);
}

// instead of having to do it like this:
{
    IXLCell cellInWorksheet = someWorksheet.Cell(someRow.RowNumber(), someColumn.ColumnNumber());
    // ^ no errors if I switch the order of these parameters

    IXLCell cellInRow = someRow.Cell(someColumn.ColumnNumber());
    IXLCell cellInColumn = someColumn.Cell(someRow.ColumnNumber());
}
```

**Comment by igitur** [comment]
I don't have convenient access to the code base right now, but would the Intersect method not solve this? I think it's on IXLRange or IXLRangeBase interface.

**Reply by airbreather** [reply]
Hm, OK, that's admittedly new to me.  Specifically looking at the "no compiler error" part of `IXLWorksheet`, it does help a bit with that:
```csharp
IXLCell cellInWorksheet = someWorksheet.Cell(someRow.Intersection(someColumn).FirstAddress);
```
but it's certainly a bit awkward and more bulky to use than `.RowNumber()` and `.ColumnNumber()`, and it trades one "no compiler error" complaint for other "no error" complaints: plenty more things implement `IXLRangeBase`, and the result of `Intersection` can be any number of cells, so fetching `.FirstAddress` from a mistaken `Intersection` call has more ways to go undetected than a more direct `.Cell(row, column)` call.

Of course, it's trivial for me to write my own extension methods (and I have), I just wanted to call this out because [this wiki page](https://github.com/ClosedXML/ClosedXML/wiki/Simplifying-your-life) lists a bunch of other implemented helpers that are even more trivial to implement than these particular extension methods would be, so it seemed fair play :grin:.
```csharp
public static IXLCell Cell(this IXLWorksheet worksheet, IXLRow row, IXLColumn column)
{
    return worksheet.Cell(row.RowNumber(), column.ColumnNumber());
}

public static IXLCell Cell(this IXLRow row, IXLColumn column)
{
    return row.Cell(column.ColumnNumber());
}

public static IXLCell Cell(this IXLColumn column, IXLRow row)
{
    return column.Cell(row.RowNumber());
}
```
